library/api/src/main/java/com/epmedu/animeal/api/feeding/FeedingApiImpl.kt

```diff
@@ -3,27 +3,50 @@ package com.epmedu.animeal.api.feeding
 import CancelFeedingMutation
 import FinishFeedingMutation
 import RejectFeedingMutation
+import SearchFeedingHistoriesQuery
 import StartFeedingMutation
 import com.amplifyframework.datastore.generated.model.Feeding
 import com.epmedu.animeal.api.extensions.getModelList
 import com.epmedu.animeal.api.extensions.launch
 import com.epmedu.animeal.common.data.wrapper.ApiResult
 import kotlinx.coroutines.flow.Flow
+import type.SearchableFeedingHistoryFilterInput
+import type.SearchableStringFilterInput
 
 internal class FeedingApiImpl : FeedingApi {
 
     override fun getUserFeedings(userId: String): Flow<List<Feeding>> {
         return getModelList(predicate = Feeding.USER_ID.eq(userId))
     }
 
+    override suspend fun getApprovedFeedingHistories(feedingPointId: String): ApiResult<SearchFeedingHistoriesQuery.Data> {
```

Comment:
Since current screen is showing just the latest feeder it would be good to grab just a short amount of these.

Reply:
Unfortunately, we can't limit them, since I had to remove sort parameter from all search queries, because of some issues with apollo generation, and search queries don't sort items as we need.